app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -158,10 +163,19 @@ def set_pre_prov_vars
     @edit[:vm_sortcol] ||= "name"
     @edit[:prov_type] = "VM Provision"
     @edit[:hide_deprecated_templates] = true if request.parameters[:controller] == "vm_cloud"
+
     unless %w(image_miq_request_new miq_template_miq_request_new).include?(params[:pressed])
+      report_name = request.parameters[:controller] == "vm_cloud" ? "ProvisionCloudTemplates.yaml" : "ProvisionInfraTemplates.yaml"
```

Comment:
This looks wrong, the same condition is there twice and we're essentially duplicating the logic already in `get_template_kls`, except differently.

Not sure `get_template_kls` is really right, that fallback to infra whenever a `prov_type` param is there sounds ..surprising.

But there is no reason to introduce these kinds of hardcoded special cases.

This should help:

```diff
diff --git a/app/controllers/application_controller/miq_request_methods.rb b/app/controllers/application_controller/miq_request_methods.rb
index d4c96013d..49c93fe65 100644
--- a/app/controllers/application_controller/miq_request_methods.rb
+++ b/app/controllers/application_controller/miq_request_methods.rb
@@ -143,8 +143,7 @@ module ApplicationController::MiqRequestMethods
   def render_updated_templates
     report_scopes = [:eligible_for_provisioning]
     report_scopes.push(:non_deprecated) if @edit[:hide_deprecated_templates]
-    report_name = request.parameters[:controller] == "vm_cloud" ? "ProvisionCloudTemplates.yaml" : "ProvisionInfraTemplates.yaml"
-    options = options_for_provisioning(get_template_kls.to_s, report_scopes, report_name)
+    options = options_for_provisioning(get_template_kls.to_s, report_scopes)
 
     @report_data_additional_options = ApplicationController::ReportDataAdditionalOptions.from_options(options)
     @report_data_additional_options.with_no_checkboxes(true)
@@ -165,12 +164,11 @@ module ApplicationController::MiqRequestMethods
     @edit[:hide_deprecated_templates] = true if request.parameters[:controller] == "vm_cloud"
 
     unless %w(image_miq_request_new miq_template_miq_request_new).include?(params[:pressed])
-      report_name = request.parameters[:controller] == "vm_cloud" ? "ProvisionCloudTemplates.yaml" : "ProvisionInfraTemplates.yaml"
-      path_to_report = ManageIQ::UI::Classic::Engine.root.join("product", "views", report_name).to_s
+      path_to_report = ManageIQ::UI::Classic::Engine.root.join("product", "views", provisioning_report).to_s
       @view = MiqReport.new(YAML.safe_load(File.open(path_to_report), [Symbol]))
       @view.db = get_template_kls.to_s
       report_scopes = %i(eligible_for_provisioning non_deprecated)
-      options = options_for_provisioning(@view.db, report_scopes, report_name)
+      options = options_for_provisioning(@view.db, report_scopes)
 
       @report_data_additional_options = ApplicationController::ReportDataAdditionalOptions.from_options(options)
       @report_data_additional_options.with_no_checkboxes(true)
@@ -1051,12 +1049,12 @@ module ApplicationController::MiqRequestMethods
     ->(_) { true } # do not apply a filter
   end
 
-  def options_for_provisioning(db, report_scopes, report_name)
+  def options_for_provisioning(db, report_scopes)
     {
       :model         => db,
       :gtl_type      => "table",
       :named_scope   => report_scopes,
-      :report_name   => report_name,
+      :report_name   => provisioning_report,
       :custom_action => {
         :url  => "/miq_request/pre_prov/?sel_id=",
         :type => 'provisioning'
diff --git a/app/controllers/vm_cloud_controller.rb b/app/controllers/vm_cloud_controller.rb
index fe927f4df..ccac118e7 100644
--- a/app/controllers/vm_cloud_controller.rb
+++ b/app/controllers/vm_cloud_controller.rb
@@ -243,6 +243,10 @@ class VmCloudController < ApplicationController
     breadcrumb_prohibited_for_action?
   end
 
+  def provisioning_report
+    'ProvisionCloudTemplates.yaml'
+  end
+
   menu_section :clo
   has_custom_buttons
 end
diff --git a/app/controllers/vm_infra_controller.rb b/app/controllers/vm_infra_controller.rb
index 4a7818386..4b8263a58 100644
--- a/app/controllers/vm_infra_controller.rb
+++ b/app/controllers/vm_infra_controller.rb
@@ -71,6 +71,10 @@ class VmInfraController < ApplicationController
     breadcrumb_prohibited_for_action?
   end
 
+  def provisioning_report
+    'ProvisionInfraTemplates.yaml'
+  end
+
   menu_section :inf
   has_custom_buttons
 end
```

(unless there is yet another controller which shows the same list, in which case we'll at least get an error instead of silently falling back to infra)